flink-end-to-end-tests/test-scripts/common_yarn_docker.sh

```diff
@@ -20,6 +20,7 @@ set -o pipefail
 
 source "$(dirname "$0")"/common.sh
 source "$(dirname "$0")"/common_docker.sh
+source "${END_TO_END_DIR}/download_artifacts.sh"
```

Comment:
I would suggest to locate the script next to `common.sh`. Maybe also rename it to common_caching_artifact_downloader.sh ? (or a nicer name -- Germans like long names)